cidc_api/models/models.py

```diff
@@ -330,6 +335,58 @@ class Permissions(CommonColumns):
     trial_id = Column(String, nullable=False, index=True)
     upload_type = Column(String, nullable=False)
 
+    @with_default_session
+    def insert(self, session: Session, commit: bool = True, compute_etag: bool = True):
+        """
+        Insert this permission record into the database and add a corresponding IAM policy binding
+        on the GCS data bucket.
+
+        NOTE: values provided to the `commit` argument will be ignored. This method always commits.
+        """
+        grantee = Users.find_by_id(self.granted_to_user)
+        if grantee is None:
+            raise IntegrityError(
+                params=None,
+                statement=None,
+                orig=f"`granted_to_user` user must exist, but no user found with id {self.granted_to_user}",
+            )
+
+        # Grant IAM permission in GCS
+        grant_download_access(grantee.email, self.trial_id, self.upload_type)
```

Comment:
I think this should go after `db.insert` so we won't have hangin IAM perms.

Reply:
Wait, how is this resolved? 

imagine session fails and rolls back, but `revoke_download_access` fails as well for whatever reason. In that scenario we end up with: IAM still granted but we don't have no corresponding record.